### PR TITLE
Trying to import a file exceeding size limit now displays a friendly error message before sending request to backend

### PR DESCRIPTION
**Pull request by klesaulnier**

### CODE REVIEW

**Comment by JacquesBorsenberger** [comment]
you mixed 2 PR ;) you should create a new branch for main, and cherry-pick your commit

**Comment by klesaulnier** [comment]
> you mixed 2 PR ;) you should create a new branch for main, and cherry-pick your commit
Fixed by editing the PR to be compared with the right branch

**Review by JacquesBorsenberger — APPROVED** [review]

File: src/translations/en.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "caseName": "case name",
     "uploadCase": "Upload case",
     "uploadErrorMsg": "Please upload the case file",
+    "uploadFileExceedingLimitSizeErrorMsg": "Select case exceeds size limit ({maxSize} MB).{br}Please compress the file in zip, gzip or bzip2 format then try to upload it again",
```
Comment (sBouzols) [inline src/translations/en.json]: Maybe use 'file' instead of 'case' here for the generic message

File: src/translations/fr.json [hunk 2]
```diff
@@ -30,6 +30,7 @@
     "caseName": "nom de la situation",
     "uploadCase": "télécharger la situation",
     "uploadErrorMsg": "Veuillez télécharger la situation",
+    "uploadFileExceedingLimitSizeErrorMsg": "La situation sélectionnée dépasse la taille limite ({maxSize} Mo).{br}Veuillez compresser ce fichier au format zip, gzip ou bzip2 puis réessayez de l'importer à nouveau.",
```
Comment (sBouzols) [inline src/translations/fr.json]: -> "le fichier sélectionné"

File: src/components/dialogs/field-hook.js [hunk 3]
```diff
@@ -78,15 +78,45 @@ export const useTextValue = ({
...
+            setFileError(
+                fileExceedsLimitMessage
+                    ? fileExceedsLimitMessage
+                    : intl.formatMessage(
+                          {
+                              id: 'uploadFileExceedingLimitSizeErrorMsg',
+                          },
+                          {
+                              maxSize: MAX_FILE_SIZE_IN_MO,
+                              br: <br />,
```
Comment (sBouzols) [inline src/components/dialogs/field-hook.js:112]: I'm not sure to pass <br/> node here is a good practice... is it already used somewhere else ?
Maybe replace this with \n directly in the translation file ?

File: src/components/dialogs/field-hook.js [hunk 4]
```diff
@@ -78,15 +78,45 @@ export const useTextValue = ({
...
+export const useFileValue = ({ active, fileExceedsLimitMessage }) => {
     const selectedFile = useSelector((state) => state.selectedFile);
+    const intl = useIntl();
     const dispatch = useDispatch();
+    const [isFileOk, setIsFileOk] = useState(false);
+    const [fileError, setFileError] = useState();
 
     const field = <UploadCase />;
     useEffect(() => {
         dispatch(removeSelectedFile());
```
Comment (sBouzols) [inline src/components/dialogs/field-hook.js:90]: It's not he role of this custom hook to set the redux state.selectedFile but it manages its reset in this useEffect. Seems strange behavior for me.

**Review by sBouzols — APPROVED** [review]
Code Review ok (some minor comments)
No new console warnings
Tests done :
* Case importation dialog
* Study creation dialog
* file with error then file without error

File: src/translations/en.json (hunk 1 above)
Comment (klesaulnier) [reply src/translations/en.json]: Fixed

File: src/translations/fr.json (hunk 2 above)
Comment (klesaulnier) [reply src/translations/fr.json]: Fixed

File: src/components/dialogs/field-hook.js (hunk 4 above)
Comment (klesaulnier) [reply src/components/dialogs/field-hook.js:90]: will be fixed later

File: src/components/dialogs/field-hook.js (hunk 3 above)
Comment (sBouzols) [reply src/components/dialogs/field-hook.js:112]: ok already done like that somewhere else then ok

**Pull request merged**